### PR TITLE
Use tudatpy conda package to build docs on Read the Docs

### LINKED ISSUE #174: Use tudatpy conda package to build docs on Read the Docs
This is a follow up to the changes introduced via #157. 

After pull request #157, tudat-multidoc is no longer needed to build the API documentation locally, however, read the docs continues to use multidoc to build the docs as evident from the body of the if loop in docs/src/conf.py. 

Modify the steps to build the docs on Read the Docs to use tudatpy conda package instead of multidoc. Point to the location where the binaries are stored, i.e., the location of the installation of tudatpy conda package on the Read the Docs server.

### PR DESCRIPTION
**Pull request by niketagrawal**
This is a follow up to the changes introduced via #157 
After pull request #157, tudat-multidoc is no longer needed to build the API documentation locally, however, read the docs continues to use multidoc to build the docs as evident from the body of the if loop in docs/src/conf.py.

Closes #174.

### Summary of changes
- docs.source/conf.py: Modify the steps to build the docs on Read the Docs to use tudatpy conda package instead of multidoc. Point to the location where the binaries are stored, i.e., the location of the installation of tudatpy conda package on the Read the Docs server.
- .readthedocs.yaml: Read the docs should check out the latest tag instead of local commit to remain consistent with the source it uses to build the docs, i.e., the tudatpy conda package. Use pre-install step to override the default build process. 
- environment_readthedocs.yml: Unpin the tudatpy version to get the latest version. Remove the dependencies that will be installed as part of tudatpy installation such as tudat, boost-cpp, numpy, matplotlib. Pinning Python=3.8 causes the tudatpy dev version to be restricted to max 0.8.0.dev11. This version pinning of python is removed. 

### Testing
[Docs successfully built and rendered by Read the Docs](https://py.api.tudat.space/en/174-tudatpy-condapkg-for-rtd-docs-build/)

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -15,6 +15,30 @@ build:
   os: "ubuntu-20.04"
   tools:
     python: "mambaforge-4.10"
+  jobs:
+    pre_install:
+      - LATEST_TAG=$(git describe --tags --abbrev=0) ; echo Downloading $LATEST_TAG ;
+       wget https://github.com/tudat-team/tudatpy/archive/refs/tags/$LATEST_TAG.zip; unzip $LATEST_TAG.zip;
+       mv docs/environment_readthedocs.yaml /tmp/environment_readthedocs.yaml;
```
Comment (yiquintero) [inline .readthedocs.yml:22]: Remove line 22 and 24. These lines were added for testing. In production we want to use the docs/enviornment_readthedocs.yaml from the latest tag in the tudatpy develop branch, which corresponds to the latest dev conda package release.

File: .readthedocs.yml [hunk 2]
```diff
@@ -15,6 +15,30 @@ build:
   os: "ubuntu-20.04"
   tools:
     python: "mambaforge-4.10"
+  jobs:
+    pre_install:
+      - LATEST_TAG=$(git describe --tags --abbrev=0) ; echo Downloading $LATEST_TAG ;
+       wget https://github.com/tudat-team/tudatpy/archive/refs/tags/$LATEST_TAG.zip; unzip $LATEST_TAG.zip;
+       mv docs/environment_readthedocs.yaml /tmp/environment_readthedocs.yaml;
+       rm -rf docs/; LT=$(echo $LATEST_TAG | cut -c 2-); mv tudatpy-$LT/docs/ .;
+       mv /tmp/environment_readthedocs.yaml docs/environment_readthedocs.yaml;
```
Comment (yiquintero) [inline .readthedocs.yml:24]: Remove line 24. See comment above.

File: .readthedocs.yml (hunk 1 above)
Comment (niketagrawal) [reply .readthedocs.yml]: Lines 22 and 24 are now removed.
Thanks.

File: .readthedocs.yml (hunk 2 above)
Comment (niketagrawal) [reply .readthedocs.yml]: Lines 22 and 24 are now removed.

**Pull request merged**